### PR TITLE
[Rich text] Change rich text section heading type

### PR DESCRIPTION
**Pull request by ludoboludo**
### PR Summary: 

Change the richtext section heading block type to be an `inline_richtext` type instead of `richtext`. 

Move the `component-rte.css` styling into `base.css` since its loaded on every page. 

Fixes https://github.com/Shopify/dawn/issues/2209

### Why are these changes introduced?

Continue the work done by @kjellr so that most headings in our theme are using this type.

### Visual impact on existing themes

There is a visual change expected for merchants that used the richtext specific features in their heading but we're planning to do [some backend work](https://github.com/Shopify/shopify/pull/395536) to limit the loss of their content when upgrading to the next theme version.
Though some content can still be lost like metafields 

### Testing steps/scenarios

- [ ] Check out the richtext section and make sure it's an `inline_richtext` allowing you to only do bold, italic and links. 
- [ ] Make sure there isn't any other place in which content should have been tweaked. 

⚠️ There has been some formatting changes due to our liquid prettier. They don't change anything but the format only 

### Demo links

- [Editor](https://os2-demo.myshopify.com/admin/themes/139430658070/editor)

### Checklist
- [ ] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [ ] Requested review from UX (Only for changes that are affecting the experience or perceivable visual details)
- [ ] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

File: sections/featured-product.liquid [hunk 1]
```diff
@@ -284,9 +283,19 @@
...
+                  update_url: false
+                %}
               {%- when 'buy_buttons' -%}
-                {%- render 'buy-buttons', block: block, product: product, product_form_id: product_form_id, section_id: section.id -%}
+                {%- render 'buy-buttons',
+                  block: block,
+                  product: product,
+                  product_form_id: product_form_id,
+                  section_id: section.id
+                -%}
```
Comment (kjellr) [inline sections/featured-product.liquid:298]: There are a bunch of little reformatting additions like these that make it harder to review. None of it appears to actually change anything though.

Comment (ludoboludo) [reply sections/featured-product.liquid:298]: Yeah 😞 I will add a note in the PR description to mention this. We're trying to default to use our own prettier extension  and I guess some stuff wasn't formatted in some files.

**Review by melissaperreault — APPROVED** [review]
Thank you Ludo! Works well! 🚢

**Review by kmeleta — APPROVED** [review]
LGTM.

**Review by kjellr — APPROVED** [review]

**Review by eugenekasimov — APPROVED** [review]
🚀

**Pull request merged**